### PR TITLE
Create `dex_solana.sandwiches` & `dex_solana.sandwiched`

### PR DESCRIPTION
**Pull request by hildobby**
### Contribution type
Please check the type of contribution this pull request is for:

- [ ] New spell(s)
- [ ] Adding to existing spell lineage
- [ ] Bug fix

**Note:** You can safely discard any section below which doesn't apply based on selection above

---

### For new spell(s)
If you are building new spell(s), please provide the following information:

- **Spell name(s):** *schema.table_name*
- **Description:** [Detailed description of the new spell(s) and their purpose]
- **Who are the new spell(s) for?** [Internal team or general community]
- **How will the new spell(s) be used downstream?** [Description of downstream usage]
- **Implementation details:** [Information on how the spell(s) are implemented]
- **Test instructions:** [How to test the new spell(s)]
- **Related issue(s):** [Link to related issues, if any]

---

### For adding to existing spell lineage
If you are adding to an existing spell lineage, please provide the following information:

- **Description:** [Description of the changes made]

---

### For bug fixes
If you are fixing a bug, please provide the following information:

- **Description:** [Description of the bug fix]
- **Steps to reproduce:** [How to reproduce the bug]
- **Implementation details:** [Information on how the bug was fixed]
- **Test instructions:** [How to test the fix]
- **Related issue(s):** [Link to related issues, if any]

---

### Additional information
Please provide any additional information that might help us review your pull request:

- [Any additional information]

---

Thank you for your contribution!

### CODE REVIEW

**Comment by 0xRobin** [comment]
Note the newly created Solana specific dbt subproject where these models should move to: https://github.com/duneanalytics/spellbook/pull/6197
Also it seems we don't need macro's for these models as they are only used for Solana (and probably won't be compatible with other chains?)

**Comment by hildobby** [comment]
> Note the newly created Solana specific dbt subproject where these models should move to: #6197 Also it seems we don't need macro's for these models as they are only used for Solana (and probably won't be compatible with other chains?)

you guys should inform us when stuff is moving to a diff dbt subprojects, not ideal to have to rework stuff on every single one of my PRs these days (only reason i have many open prs is because i cant work on one at a time given i need to wait for ci runs to finish since i cant test stuff locally + rerun them all for test tables on every new working day)

i used macros here cause it was easier to port but sure i can make it a normal spell instead. would we be able to get macros working across dbt projects? even if not natively capable, there has to be a way to clone necessary yml files and macros across dbt subprojects, right?

**Comment by aalan3** [comment]
@hildobby sorry about the mess, we are getting close to being done. 

Regarding macros the top level `macros` folder is shared across all projects. There are also sources that are shared across all projects. In this case I don't think having this as a shared macro would change anything but maybe you are referring to something else?

**Comment by hildobby** [comment]
> @hildobby sorry about the mess, we are getting close to being done.
> 
> Regarding macros the top level `macros` folder is shared across all projects. There are also sources that are shared across all projects. In this case I don't think having this as a shared macro would change anything but maybe you are referring to something else?

oh this wasn't about this macro specifically but about stuff being available across subprojects, what folders exactly are shared across dbts? is it only macros? sources would also be cool for example

**Comment by 0xRobin** [comment]
> oh this wasn't about this macro specifically but about stuff being available across subprojects, what folders exactly are shared across dbts? is it only macros? sources would also be cool for example

currently sources and macro's are shared across projects! 
You can see this in the `dbt_project.yml` file of the subprojects, they will list the `../sources` and `../macros` directories.

**Comment by hildobby** [comment]
> > oh this wasn't about this macro specifically but about stuff being available across subprojects, what folders exactly are shared across dbts? is it only macros? sources would also be cool for example
> 
> currently sources and macro's are shared across projects! You can see this in the `dbt_project.yml` file of the subprojects, they will list the `../sources` and `../macros` directories.

oh ok so its just that it's refs where it's built and needs to be defined as sources when used somewhere else? maybe all sources should be moved to their own folder that is shared across all projects then?

**Review by andrewhong5297 — APPROVED** [review]
lgtm!

**Comment by jeff-dude** [comment]
closing due to inactivity. feel free to reopen and describe what needs to happen next, if necessary. thank you.

File: dbt_subprojects/solana/models/_sector/dex/sandwiches/solana/dex_solana_sandwiches.sql
Comment (jeff-dude) [inline dbt_subprojects/solana/models/_sector/dex/sandwiches/solana/dex_solana_sandwiches.sql]: @andrewhong5297 i was able to remove the join to this raw table, as solana dex has the two fields requested from what i can tell so i simply added in selects

**Comment by jeff-dude** [comment]
@andrewhong5297 plz give final review with latest changes 🙏

**Comment by andrewhong5297** [comment]
Will need @hildobby to review the cte changes when he back from OOO

**Comment by jeff-dude** [comment]
> Will need @hildobby to review the cte changes when he back from OOO

sounds good, will leave in queue until approved

**Comment by hildobby** [comment]
> Will need @hildobby to review the cte changes when he back from OOO

lgtm @andrewhong5297, just added you as a contributor to the spell

cc @jeff-dude

**Pull request merged**
